### PR TITLE
Add remove_duplicates.py to Ch.11

### PR DESCRIPTION
**Pull request by srikar-eranky**
Srikar Eranky

**Remove_duplicates.py**
* Chapter 11
* Practice

Added a couple hints to the practice template for sorting and counting.

### CODE REVIEW

**Review by Citrus716 — CHANGES_REQUESTED** [review]
Nice Problem, but I don't think the solution works all the time. I tried the list: [2, 2, 1, 2, 324, 52, 34, 324], and I got [1, 2, 2, 34, 52, 324, 324]. Let me know if you need any help.

**Review by abhatia1205 — CHANGES_REQUESTED** [review]
Hi Srikar.

Generally, it is bad to change a list while you are iterating over it. It messes up the loop and often leads to errors, which has happened in your case. For example, if I try the list [1,1,1,1,1,1,1], it returns [1,1], which is incorrect. I would suggest changing the code such that you are not changing the list while iterating. Having a worse space complexity by making a new list is fine because iterating and changing the list at the same time is usually bad practice.

Another note: You should make sure to specify that using the set function is NOT ALLOWED. If it was allowed, the function would just be one line of code: return list(set(array)).

Please make these changes and resubmit. Thanks!

File: 2_intermediate/chapter11/practice/remove_duplicates.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Write a function called remove_duplicates
+# The sole parameter of the function should be a list
+# The function should look through a list,
+# Find all duplicate elements, and remove them
+# Sort the resulting list
```
Comment (abhatia1205) [inline 2_intermediate/chapter11/practice/remove_duplicates.py:5]:
```suggestion
# Sort the resulting list
# YOU MAY NOT USE THE set() function IN PYTHON. Use this problem to practice list iteration!
```

File: 2_intermediate/chapter11/solutions/remove_duplicates.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Write a function called remove_duplicates
+# The sole parameter of the function should be a list
+# The function should look through a list,
+# Find all duplicate elements, and remove them
+# Sort the resulting list
```
Comment (abhatia1205) [inline 2_intermediate/chapter11/solutions/remove_duplicates.py:5]:
```suggestion
# Sort the resulting list
# YOU MAY NOT USE THE set() function IN PYTHON. Use this problem to practice list iteration!
```

File: 2_intermediate/chapter11/solutions/remove_duplicates.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# Example: array = [1,1,2,5,4,6,12,3,4,6]
+# Result should print [1,2,3,4,5,6,12]
+
+# Write code here
+
+list1 = [1, 1, 2, 5, 4, 6, 12, 3, 4, 6]  # Define your list
+
+
+def remove_duplicate(array):
+    for i in array:
```
Comment (abhatia1205) [inline 2_intermediate/chapter11/solutions/remove_duplicates.py]: It's possible, even probable, that Python gets all the elements in ```array``` only once, which is why when you call ```array.remove()```, ```array``` changes but what the loop is iterating over does not, which causes your solution to crash.

Comment (srikar-eranky) [reply 2_intermediate/chapter11/solutions/remove_duplicates.py]: Hmm...I'll make a new solution and check if it works, and then I'll fix it. Thanks for the input.

**Comment by abhatia1205** [comment]
Ill just go ahead and merge this. Formatting seems ok.

**Pull request merged**